spec/controllers/catalog_controller_spec.rb

```diff
@@ -654,45 +603,41 @@ def export_as_mock
 
   describe "#add_show_tools_partial" do
     before do
-      CatalogController.add_show_tools_partial(:like, callback: :perform_like, validator: :validate_like_params)
-      allow(controller).to receive(:perform_like)
+      described_class.add_show_tools_partial(:like, callback: :perform_like, validator: :validate_like_params)
       allow(controller).to receive(:solr_document_url).and_return('catalog/1')
-      allow(controller).to receive(:action_documents).and_return([SolrDocument.new])
+      allow(controller).to receive(:action_documents).and_return(1)
       Rails.application.routes.draw do
         get 'catalog/like', as: :catalog_like
       end
     end
 
     after do
-      CatalogController.blacklight_config.show.document_actions.delete(:like)
+      described_class.blacklight_config.show.document_actions.delete(:like)
       Rails.application.reload_routes!
     end
 
     it "adds the action to a list" do
-      expect(CatalogController.blacklight_config.show.document_actions).to have_key(:like)
+      expect(described_class.blacklight_config.show.document_actions).to have_key(:like)
     end
 
     it "defines the action method" do
-      expect(controller.respond_to?(:like)).to be true
+      expect(controller).to respond_to(:like)
     end
 
     describe "when posting to the action" do
-      describe "with success" do
-        before do
-          allow(controller).to receive(:validate_like_params).and_return(true)
-          post :like
-        end
+      context 'with valid params' do
         it "calls the supplied method on post" do
-          expect(controller).to have_received(:perform_like) 
+          expect(controller).to receive(:validate_like_params).and_return(true)
+          expect(controller).to receive(:perform_like)
+          post :like
         end
       end
 
-      describe "with failure" do
-        describe "with invalid params" do
-          before { allow(controller).to receive(:validate_like_params).and_return(false) }
-          it "does not call the supplied method if validation failed" do
-            expect(controller).not_to have_received(:perform_like)
-          end
+      context "with failure on invalid params" do
+        it "does not call the supplied method" do
+          expect(controller).to receive(:validate_like_params).and_return(false)
+          expect(controller).not_to receive(:perform_like)
+          expect { post :like }.not_to raise_error
```

Comment:
This is now the sole failing test on Rails 4.  

Note: the old version of the test was bogus.  The post was already made at 683 _before_ `validate_like_params` was re-stubbed false.